packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -110,6 +111,17 @@ Future<Depfile> copyAssets(
 
   await Future.wait<void>(
     assetEntries.entries.map<Future<void>>((MapEntry<String, AssetBundleEntry> entry) async {
+      // Determine the current build target (e.g. "windows", "android", "ios")
+      // and compare it against the asset's declared `platforms` set from
+      // pubspec.yaml. If the asset specifies one or more platforms and the
+      // current target is not included, skip processing this asset to ensure
+      // it is not bundled for the wrong platform.
+      final String target = targetPlatform.osName;
+      final Set<String> targetPlatforms = entry.value.platforms;
+      if (targetPlatforms.isNotEmpty && !targetPlatforms.contains(target)) {
```

Comment:
Is there any reason why we're doing the filtering here, instead of in the `AssetBundle.build()` function like how `flavors` is implemented?

If it's done there, we don't need to add the `platforms` entry in the `AssetBundleEntry` class.